pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,133 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
+
+
+def _lower_poa_global_limit_lorenz(solar_zenith, dni_extra):
+    r"""Function to calculate the lower limit of GTI
+    """
+    # Setting the lower_limit at 0.
+    lower_limit = pd.Series(np.zeros(len(solar_zenith)),
+                            index=solar_zenith.index)
+
+    # Determining the lower limit when solar zenith is < 75
+    lower_limit = lower_limit.mask(solar_zenith < 75,
+                                   0.01 * dni_extra * cosd(solar_zenith))
+
+    # Setting upper limit as undefined where solar_zenith is not available
+    lower_limit[solar_zenith.isna()] = np.nan
+
+    # Renaming upper_limit series to 'upper_limit'
+    lower_limit.rename('lower_limit')
+    return (lower_limit)
+
+
+def check_poa_global_limits_lorenz(poa_global, solar_zenith, aoi):
+    r"""Test for limits on POA global using the equations described in
+    Section 6.1 of [1]_
+
+    Criteria from [1] are used to determine physically plausible
+    lower and upper bounds. Each value is tested and a value passes if
+    value > lower bound and value < upper bound. Lower bounds are
+    constant for all tests. Upper bounds are calculated as
+
+    .. math::
+        ub = min + mult * dni\_extra * cos( solar\_zenith)^{exp}
+
+    Parameters
+    ----------
+    poa_global : Series
+        Global tilted irradiance in :math:`W/m^2`
+    solar_zenith : Series
+        Solar zenith angle in degrees
+    aoi : Series
+        Direct normal irradiance in :math:`W/m^2`
+
+    Returns
+    -------
+    poa_global_limit_bool_flag : Series
+        True for each value that is physically possible.
+    poa_global_limit_int_flag : Series
+        Series of integers representing the flag numbers described in the
+        literature [1]
+
+    Notes
+    -----
+    The upper limit for `poa_global` is set to 0 when `solar_zenith` is greater
+    than 90 degrees. Missing values of `poa_global`, `solar_zenith`
+    and/or `aoi` will result in a `False` flag.
+
+    References
+    ----------
+    .. [1] Elke Lorenz et. al, High resolution measurement network of global
+           horizontal and tilted solar irradiance in southern Germany with a
+           new quality control scheme, Solar Energy, Volume 231, 2022,
+           Pages 593-606, ISSN 0038-092X,
+           https://doi.org/10.1016/j.solener.2021.11.023.
+    """
+    # Defining the normal irradiance at the top of atmosphere in W/m^2
+    dni_extra = 1367
+
+    # Making sure that the input are in series
+    poa_global = pd.Series(poa_global)
+    aoi = pd.Series(aoi)
+    solar_zenith = pd.Series(solar_zenith)
+
+    # Finding the upper and lower limit
+    upper_limit = _upper_poa_global_limit_lorenz(aoi, solar_zenith, dni_extra)
+    lower_limit = _lower_poa_global_limit_lorenz(solar_zenith, dni_extra)
+
+    # Initiating a poa_global_limit_int_flag series
+    poa_global_limit_int_flag = \
+        pd.Series(np.zeros(len(solar_zenith)).astype(np.int64),
+                  index=solar_zenith.index)
```

Comment:
```suggestion
    poa_global_limit_int_flag = pd.Series(0, index=solar_zenith.index)
```
Passing an integer `0` results in `int64` dtype by default

Reply:
Wow! I successfully managed to make it more complicated than it should have been.